codex/blockexchange/network/networkpeer.nim

```diff
@@ -21,6 +24,8 @@ const
   MaxMessageSize = 100 * 1 shl 20 # manifest files can be big
 
 type
+  ConnProvider* = proc(): Future[Connection] {.gcsafe, closure.}
```

Comment:
`ConnProvider` was removed in https://github.com/status-im/nim-libp2p/pull/749.

By adding it here, the types line up and compilation succeeds. However, with the behavioral changes in https://github.com/status-im/nim-libp2p/pull/749 adding the type here is probably not the thing to do, and may be responsible for failures in network related tests when running `make test`.

I spent some time looking at https://github.com/status-im/nim-libp2p/pull/749 but it wasn't entirely clear to me (yet) what the proper changes would be in nim-codex. Hopefully @Menduist can provide some insights.

Reply:
My bad, I didn't realized this type was used outside of libp2p, should have checked
However your fix seems correct to me
The tests are probably failing for another reason. I'll take a look